packages/react-core/src/next/components/Dropdown/DropdownItem.tsx

```diff
@@ -16,6 +16,22 @@ export interface DropdownItemProps extends Omit<MenuItemProps, 'ref'>, OUIAProps
   ouiaSafe?: boolean;
 }
 
+type ConditionalDropdownItemProps =
+  | {
+      /** Target navigation link */
+      to?: string;
+      /** Flyout menu. Disallowed if nav link is defined */
+      flyoutMenu?: never;
+    }
+  | {
+      /** Target navigation link. Disallowed if flyoutMenu is defined */
+      to?: never;
+      /** Flyout menu */
+      flyoutMenu?: React.ReactElement;
+    };
+
+export type DropdownItemProps = CommonDropdownItemProps & ConditionalDropdownItemProps;
+
```

Comment:
Had to make this change so that the props would agree with `MenuItemProps`. Is there a better way to achieve this?